internal/cli/run/run.go

```diff
@@ -13,6 +16,20 @@ import (
 	"github.com/ooni/probe-cli/nettests"
 )
 
+// listenForSignals will listen for SIGINT and SIGTERM. When it receives those
+// signals it will set isTerminated to true, which will cleanly shutdown the
+// test logic
```

Comment:
Please, document that the proper way to attain this objective would instead be to use a `context.Context` and cancel it when a signal is delivered, but this kind of refactoring is too complex for now. It would perhaps be top to have an issue that suggests to use a cancellable context rather than just adding a TODO comment to the code.